src/Controller/Traits/LoginTrait.php

```diff
@@ -218,14 +218,7 @@ public function verify()
             return $this->redirect(Configure::read('Auth.loginAction'));
         }
 
-        // storing user's session in the temporary one
-        // until the GA verification is checked
-        $temporarySession = $this->Auth->user();
-        $this->request->getSession()->delete('Auth.User');
-
-        if (!empty($temporarySession)) {
-            $this->request->getSession()->write('temporarySession', $temporarySession);
-        }
+        $temporarySession = $this->request->getSession()->read('temporarySession');
 
         if (array_key_exists('secret', $temporarySession)) {
```

Comment:
It probably won't interfer with authentication anymore, but this will still trigger an error in case `temporarySession` isn't set, or if it contains a non-array for whatever reason.